### DISCUSSION TITLE
Releases from master?

### DISCUSSION THREAD

**Discussion by platy** [diesel-rs/diesel#2761] [Q&A] [answered]
I'm suffering from #2203 and #2071 and was happy to see that these issues are both fixed months ago on master. I upgraded to 1.4.6 but that didn't help as there are "1204 commits to master since this release".

What is the general plan for releases from master? I've been looking for an explanation but haven't found one. It might be handy if it was explained on the readme.

Should I be using master instead of 1.4.x?
Could these fixes be patched into the 1.4.x branch? (I could make the PR if so)
Is there a roadmap for 2.0?

**Comment by weiznich** [comment, marked as answer]
> I'm suffering from #2203 and #2071 and was happy to see that these issues are both fixed months ago on master. I upgraded to 1.4.6 but that didn't help as there are "1204 commits to master since this release".
>
>What is the general plan for releases from master? I've been looking for an explanation but haven't found one. It might be handy if it was explained on the readme.

Master tracks the next feature release, as this is common in the rust ecosystem (and likely nearly everywhere else). As the next release is planed to be a major version bump we try to batch as many breaking changes as we see happening in the next time into this release. That means it take some time to implement all this stuff, especially if you only work on this project in your free time. In In addition to that this means we do not have the capacity to backport features that don't need the breaking changes in 2.0 to an older release as this would need even more resources. We plan to release all new features someday as 2.0, but we cannot give you an ETA. If you want this to happen sooner consider contributing something or found someone to do this for you.

> Should I be using master instead of 1.4.x?

That's up to you, but keep in mind that we don't guarantee anything there (though we are happy to receive bug reports and general reviews of the design). Additionally things can and will break there between versions.

> Could these fixes be patched into the 1.4.x branch? (I could make the PR if so)

#2203 does not happen with any release, as this depends on the caching that is only available in the master branch. I consider the caching itself as new feature, not as bug fix. Therefore this is nothing that should be released as patch release.

> Is there a roadmap for 2.0?

See the [corresponding milestone](https://github.com/diesel-rs/diesel/milestone/16) for list of open issues we want to address before the next release. As this is a free time project we will not give any timeline about when this work will be finished. That release will likely come faster if more people start contributing something on that list.

**Reply by platy** [reply]
Thanks for the clarifications. I guess I was surprised by the scale of the next feature release, but everything you are saying makes sense. I'm sorry I failed to find the milestones myself!

I understand why #2071 caching isn't a patch release thing, I think I missed before as I only discovered it through #2203, although that is still a bug with or without caching. Would you be interested in a 1.5 feature release (and a 1.x feature branch) if I managed to backport caching? I know it doesn't help getting to the next milestone but it would get me introduced to some diesel internals. And it could be useful to others.

**Reply by weiznich** [reply]
Yes #2203 can happen without the caching, but fixing this requires a "new" feature that allows you to specify the schema. I don't want to introduce that as patch release.

Doing a 1.5 release is not something we want to do, because this would raise quite a lot questions which features exactly should be back ported and who does the work, testing and so. I would rather like to spend that time on finishing the 2.0 release.

**Reply by platy** [reply]
Understood, thanks
